src/tracer.cc

```diff
@@ -338,18 +457,21 @@ TraceManager::TraceActivity(
   auto ts =
       reinterpret_cast<std::shared_ptr<TraceManager::Trace>*>(userp)->get();
 
+
   std::lock_guard<std::mutex> lk(ts->mtx_);
   std::stringstream* ss = nullptr;
   {
-    if (ts->streams_.find(id) == ts->streams_.end()) {
-      std::unique_ptr<std::stringstream> stream(new std::stringstream());
-      ss = stream.get();
-      ts->streams_.emplace(id, std::move(stream));
-    } else {
-      ss = ts->streams_[id].get();
-      // If the string stream is not newly created, add "," as there is
-      // already content in the string stream
-      *ss << ",";
+    if (ts->setting_->mode_ == TRACE_MODE_TRITON) {
```

Comment:
I think its OK for now to get into release, but I think `TraceActivity` and `TraceTensorActivity()` should be refactored with helper functions, something like below. I find the current function pretty hard to parse through, lots of `if`s and `ifdef` to parse:
```
    if (ts->setting_->mode_ == TRACE_MODE_TRITON) {
      TraceActivityTriton()
    } 

    else if (OPENTELEMETRY) {
      TraceActivityOpenTelemetry();
    }
    
    // etc.
```